Libraries/LibGUI/ColorPicker.cpp

```diff
@@ -28,88 +28,278 @@
+    m_manin_container->set_size_policy(SizePolicy::Fill, SizePolicy::Fill);
+    m_manin_container->set_layout<VerticalBoxLayout>();
+    m_manin_container->layout()->set_spacing(4);
+
+    auto& button_container = root_container.add<Widget>();
+    button_container.set_size_policy(SizePolicy::Fill, SizePolicy::Fixed);
+    button_container.set_preferred_size(0, 22);
+    button_container.set_layout<HorizontalBoxLayout>();
+    button_container.layout()->set_spacing(4);
+    button_container.layout()->add_spacer();
+
+    m_custom_button = button_container.add<Button>();
+    m_custom_button->set_size_policy(SizePolicy::Fixed, SizePolicy::Fill);
+    m_custom_button->set_preferred_size(80, 0);
+    m_custom_button->set_text("Custom Color");
+    m_custom_button->on_click = [this] {
+        if (m_current_screen == ScreenList::Palette) {
+            this->build_ui_custom(*m_manin_container);
+        } else {
+            this->build_ui_palette(*m_manin_container);
+        }
+    };
+
+    auto& cancel_button = button_container.add<Button>();
+    cancel_button.set_size_policy(SizePolicy::Fixed, SizePolicy::Fill);
+    cancel_button.set_preferred_size(80, 0);
+    cancel_button.set_text("Cancel");
+    cancel_button.on_click = [this] {
+        done(ExecCancel);
+    };
+
+    auto& ok_button = button_container.add<Button>();
+    ok_button.set_size_policy(SizePolicy::Fixed, SizePolicy::Fill);
+    ok_button.set_preferred_size(80, 0);
+    ok_button.set_text("Select");
+    ok_button.on_click = [this] {
+        done(ExecOK);
+    };
+
+    if (m_current_screen == ScreenList::Palette) {
+        build_ui_palette(*m_manin_container);
+    } else {
+        build_ui_custom(*m_manin_container);
+    }
+}
+
+void ColorPicker::build_ui_palette(Widget& kok)
+{
+    for (auto& value : kok.child_widgets()) {
+        value->remove_from_parent();
+    }
+
+    resize(530, 160);
+    m_current_screen = ScreenList::Palette;
+    this->m_custom_button->set_text("Custom Color");
+
+    unsigned colors[4][9] = {
+        { 0xef2929, 0xf0b143, 0xfce94f, 0x9fe13a, 0x7c9ece, 0xa680a8, 0xe1ba70, 0x888a85, 0xeeeeec },
+        { 0xba1e09, 0xf57900, 0xe9d51a, 0x8bd121, 0x4164a3, 0x6f517b, 0xb77f19, 0x555753, 0xd4d7cf },
+        { 0x961605, 0xbf600c, 0xe9d51a, 0x619910, 0x2b4986, 0x573666, 0x875b09, 0x2f3436, 0xbbbdb6 },
+        { 0x000000, 0x2f3436, 0x555753, 0x808080, 0xbabdb6, 0xd3d7cf, 0xeeeeec, 0xf3f3f3, 0xffffff }
+    };
 
-    auto& left_vertical_container = horizontal_container.add<Widget>();
-    left_vertical_container.set_layout<VerticalBoxLayout>();
+    for (int r = 0; r < 4; r++) {
+        auto& colors_row = kok.add<Widget>();
+        colors_row.set_layout<HorizontalBoxLayout>();
+        colors_row.set_size_policy(SizePolicy::Fill, SizePolicy::Fill);
+
+        for (int i = 0; i < 8; i++) {
+            create_color_button(colors_row, colors[r][i]);
+        }
+    }
+}
+
+void ColorPicker::build_ui_custom(Widget& kok)
```

Comment:
Same as `build_ui_palette()`.
```suggestion
void ColorPicker::build_ui_custom(Widget& root_container)
```